pkg/reconciler/workload/heartbeat/heartbeat_manager.go

```diff
@@ -35,25 +37,26 @@ type clusterManager struct {
 
 func (c *clusterManager) Reconcile(ctx context.Context, cluster *workloadv1alpha1.WorkloadCluster) error {
 	latestHeartbeat := time.Time{}
-	for _, c := range cluster.Status.Conditions {
-		if c.LastHeartbeatTime.Time.After(latestHeartbeat) {
-			latestHeartbeat = c.LastHeartbeatTime.Time
-		}
+	if cluster.Status.LastSyncerHeartbeatTime != nil {
+		latestHeartbeat = cluster.Status.LastSyncerHeartbeatTime.Time
 	}
 	if latestHeartbeat.IsZero() {
-		// Never seen a heartbeat; for now, this means the syncer hasn't heartbeated -- in the future, this should be un-Ready.
-		return nil
-	}
-
-	if time.Since(latestHeartbeat) > c.heartbeatThreshold {
+		klog.V(5).Infof("Marking WorkloadCluster %s|%s not ready due to no heartbeat", cluster.ClusterName, cluster.Name)
+		conditions.MarkFalse(cluster,
+			workloadv1alpha1.WorkloadClusterReadyCondition,
+			workloadv1alpha1.ErrorHeartbeatMissedReason,
+			conditionsv1alpha1.ConditionSeverityWarning,
```

Comment:
this happens on startup of the syncer, no? Would not even call this a warning. It is a normal, expected state

Reply:
Normal how? The cluster isn't ready, and can't be scheduled to. Why isn't that something to call out?

Reply:
+1 to @marun's comment, this should be considered unhealthy/un-`Ready` until some heartbeat has been observed.